File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        map<char,int> count;
        int left=0;
        int n=s.length();
        int res=0;
        int maxf=0;
        for(int r=0;r<n;r++){
            count[s[r]]++;
            maxf=max(maxf,count[s[r]]);

            while((r-left+1-maxf)>k){
                
                count[s[left]]--;
                left++;
            }
            res=max(res,r-left+1);
        }
        return res;
        
    }
};

File: Data Structures & Algorithms/max-water-container/submission-1.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        n=len(heights)
        left,right=0,n-1
        volume=0
        while(left<right):
            volume=max(volume,(right-left)*min(heights[left],heights[right]))
            if(heights[left]>heights[right]):
                right-=1
            else:
                left+=1
        return volume

        